Chat settings redesign

DispatchCommit opened the pull request:
Creates a categorized, tabbed version of chat settings to accommodate more settings.
Separates large single settings file into 3 smaller sub components, one for each settings panel.
Settings overflow within the card, rather than entire card.

Changes TTS voice selection box to solo variant.
Moves all toggles to be in line.
Increase menu title bar font weight to bold.

Current Design:
![image](https://user-images.githubusercontent.com/45262335/86236389-d85c6b00-bb4e-11ea-86f6-a102e907942f.png)
![image](https://user-images.githubusercontent.com/45262335/86236432-e6aa8700-bb4e-11ea-95b4-ce5ff8ebec46.png)
![image](https://user-images.githubusercontent.com/45262335/86236447-ef9b5880-bb4e-11ea-8bdc-385e9923a941.png)

DispatchCommit committed:
Change ban to mute

DispatchCommit committed:
Tabify chat settings

Creates 3 new subcomponents for chat settings, and coverts single menu dialog into a tabbed dialog

DispatchCommit committed:
do not eager load settings

diingus commented on components/Chat/ChatMessages/ChatMessageMenu.vue:
```diff
@@ -107,7 +107,7 @@
               small
               @click="banUser"
-            >Ban</v-btn>
+            >Mute</v-btn>
```
I still think 'master ignore list' is a better name. Might not be up to modern standards though

DispatchCommit replied:
This wasn't supposed to sneak into this PR, will revisit later on.

diingus commented on components/Chat/ChatSettings/ChatSettingsPanel/ChatBetaSettingsPanel.vue:
```diff
@@ -0,0 +1,82 @@
+    class="pb-3 px-3"
+  >
+    <!-- Beta Settings Subheader -->
+    <v-subheader class="overline mb-0">
```
```suggestion

```
No other tab starts with a subheader

DispatchCommit replied:
removed.

diingus commented on components/Chat/ChatSettings/ChatSettingsPanel/ChatGeneralSettingsPanel.vue:
```diff
@@ -0,0 +1,180 @@
+      <v-switch
+        v-model="receiveMentionsInLocal"
+        :disabled="globalChat"
+        label="Recieve @'s while in local"
```
Spelling seems wierd :)

diingus replied:
Apparently 'leave review' and 'leave single comment' do the same thing

DispatchCommit replied:
fixed. In my defense, you made that label.

DispatchCommit committed:
Use dynamic components

simplify root component by using dynamic components and v-for loops.
Add a bg color to tab bar
Change style of TTS voice dropdown
Change padding to margin
Remove beta subheader

DispatchCommit committed:
learn2spell

DispatchCommit commented:
Updated preview
![image](https://user-images.githubusercontent.com/45262335/86241613-1316d100-bb58-11ea-9377-da5c04a286b4.png)

DispatchCommit committed:
change padding to margin in all files

diingus commented on components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue:
```diff
@@ -0,0 +1,256 @@
+        <!-- Enable TTS -->
+        <v-switch
+          v-model="useTts"
+          :disabled="!ttsVoicesExist"
```
Seems redundant with the `v-if` on the same condition above.

DispatchCommit replied:
true

diingus commented on components/Chat/ChatSettings/ChatSettingsPanel/ChatTtsSettingsPanel.vue:
```diff
@@ -0,0 +1,256 @@
+          </template>
+        </v-slider>
+
+        <i class="caption grey--text">note: a value of 0 disables timeout.</i>
```
Maybe all settings should have a tooltip explaining them. Not sure what the mobile equivalent would be.

DispatchCommit replied:
Should consider improving documentation all around tbh.

diingus replied:
My `Chat.vue` diagram is just missing polls :)

DispatchCommit replied:
*dies inside*

diingus reviewed (approved):
Looks good. There's a lot of boilerplate with `get()` and `set()` being mapped to store functions. Not sure if that could be factored out with clever use of the spread operator. 

Definitely a nice change, both in UX and in code, as it's no longer a mile long file with random containers keeping it all together.

DispatchCommit committed:
Save a div, ride a template.

convert v-if to v-else
